src/json/from_json.cr

```diff
@@ -65,6 +65,38 @@ def Deque.from_json(string_or_io) : Nil
   end
 end
 
+module Iterator(T)
+  def self.from_json(string_or_io)
+    FromJson(T).new(JSON::PullParser.new(string_or_io), true)
+  end
+
+  def self.new(pull : JSON::PullParser)
+    FromJson(T).new(pull, false)
+  end
+
+  class FromJson(T)
+    include Iterator(T)
+
+    def initialize(@pull : JSON::PullParser, @check_eof : Bool)
+      @pull.read_begin_array
+      @end = false
+    end
+
+    def next
+      if @end
+        stop
+      elsif @pull.kind.end_array?
+        @pull.read_next
+        @end = true
+        raise JSON::ParseException.new("Invalid JSON", *@pull.location) if @check_eof && !@pull.kind.eof?
```

Comment:
I don't think this check is necessary. If the next token isn't EOF, the YAML is malformed and the parser already notices that (at `read_next`) and raises. Try parsing `[a,b]]` with `check_eof: false` for example.

Would be great to add specs for failure behaviour, though (invalid YAML and non-matching types).

Reply:
OK